rr_computer_vision/include/endline_detection.hpp

```diff
@@ -0,0 +1,42 @@
+/** @file endline_detection.hpp
+ * @author Waleed Ahmed (w29ahmed)
+ * @author Yuchi(Allan) Zhao
+ */
+
+#ifndef ENDLINE_DETECTION_HPP
+#define ENDLINE_DETECTION_HPP
+
+#include <ros/ros.h>
+#include <image_transport/image_transport.h>
+#include <cv_bridge/cv_bridge.h>
+#include <sensor_msgs/image_encodings.h>
+#include <opencv2/opencv.hpp>
+#include <std_srvs/Trigger.h>
+// #include "msg_srv_names.hpp"
+
+class EndlineDetection
+{
+  public :
+    EndlineDetection(ros::NodeHandle);
+    void ImgCb(const sensor_msgs::ImageConstPtr&);
+    static bool compareContourAreas( std::vector<cv::Point> contour1, std::vector<cv::Point> contour2 );
+    
+  private :
+    ros::NodeHandle nh_;
+    ros::ServiceClient client_;
+    image_transport::ImageTransport it_;
+    image_transport::Subscriber test_subscriber;
+    image_transport::Publisher test_publisher;
+
+    bool detection_status;
```

Comment:
Inconsistent naming convention; please change these to be consistent with the others